### PR TITLE
Fix summary with multiple test plans and skipped tests

### PR DESCRIPTION
**Pull request by joshdholtz**
## Motivation
Fixes https://github.com/fastlane/fastlane/pull/19629#issuecomment-1013511227

The test summary printed after a run was showing wrong results if:
- Multiple test suites were being run
- Any tests were skipped

Expected: `Tests Passed: 0 failed, 955 total (12.187 seconds)`
Actual: `Tests Passed: 0 failed, 56 total (0.324 seconds)`

The issue was because the test summary line parser assumed that there was only one line summary being printed. The line looks like 👇 

```sh
Executed 8 tests, with 4 failures (0 unexpected) in 0.154 (0.155) seconds
```

However... this line gets printed multiple times after each suite gets run 👇 

```sh
Test Case '-[Tests_iOS.Tests_iOS testExample5ThatSometimesFails]' failed (0.019 seconds).                                                              
Test Suite 'Tests_iOS' failed at 2022-01-15 21:31:49.073.                                                                                              
         Executed 8 tests, with 4 failures (0 unexpected) in 0.154 (0.155) seconds                                                                     
Test Suite 'Tests iOS.xctest' failed at 2022-01-15 21:31:49.073.                                                                                       
         Executed 8 tests, with 4 failures (0 unexpected) in 0.154 (0.155) seconds                                                                     
Test Suite 'All tests' failed at 2022-01-15 21:31:49.073.                                                                                              
         Executed 8 tests, with 4 failures (0 unexpected) in 0.154 (0.156) seconds
```

There is also a case that was unaccounted when tests were skipped 👇 

```sh
Executed 56 tests, with 3 tests skipped and 0 failures (0 unexpected) in 1.019 (1.029) seconds 
```

## Description
- Introduced 2 new line matchers for when a suite is finished for "All tests"
- Introduced 1 new line matcher for "Executed with skipped tests"

**Previously**, the `testSummary` in `Parser` would get override by the latest summary which was the root cause of this issue.

**Now**, a `testSummary` will only get parsed when it follows either `Test Suite 'All tests' passed at` or `Test Suite 'All tests' failed at`. There can also be multiple summaries so this will also add all the summaries together for:
- test count
- skipped count
- failure count
- unexpected count
- time

It also adds logic now for parsing the two different variation of the "executed" lines:
- `Executed 8 tests, with 4 failures (0 unexpected) in 0.154 (0.155) seconds `
- `Executed 56 tests, with 3 tests skipped and 0 failures (0 unexpected) in 1.019 (1.029) seconds `

### CODE REVIEW

**Review by adellibovi — CHANGES_REQUESTED** [review]
Thanks @joshdholtz for the PR ☺️.
It would be great if we can add a test so we don't regress on this 🙏🏻

**Comment by joshdholtz** [comment]
On it!

File: Sources/XcbeautifyLib/Parser.swift [hunk 1]
```diff
@@ -244,12 +255,38 @@ public class Parser {
     }
 
     func parseSummary(line: String, colored: Bool) {
+        if !needToRecordSummary {
+            return
+        }
```
Comment (danieleformichelli) [inline Sources/XcbeautifyLib/Parser.swift]:
```suggestion
        guard needToRecordSummary else {
            return
        }
```

File: Tests/XcbeautifyLibTests/XcbeautifyLibTests.swift [hunk 2]
```diff
@@ -104,6 +109,87 @@ final class XcbeautifyLibTests: XCTestCase {
...
+        // First test plan
+        XCTAssertNil(parser.summary)
+        XCTAssertFalse(parser.needToRecordSummary)
+        let formatted1 = noColoredFormatted(input1)
+        XCTAssertTrue(parser.needToRecordSummary)
+        let formatted2 = noColoredFormatted(input2)
+        XCTAssertFalse(parser.needToRecordSummary)
+        XCTAssertNil(formatted1)
+        XCTAssertNil(formatted2)
+        XCTAssertNotNil(parser.summary)
```
Comment (danieleformichelli) [inline Tests/XcbeautifyLibTests/XcbeautifyLibTests.swift]: I would unwrap it here to simplify subsequent code
```suggestion
        let summary = try XCTUnwrap(parser.summary)
```

**Review by adellibovi — APPROVED** [review]
Thanks @joshdholtz , LGTM! 🥳  cc @tuist/core

**Review by danieleformichelli — APPROVED** [review]

**Comment by joshdholtz** [comment]
Any idea why two tests failed on Ubuntu? 😬

**Comment by luispadron** [comment]
> Any idea why two tests failed on Ubuntu? 😬

I’m not entirely sure but the tests that are failing are:

    ✖ testExecuted, XCTAssertTrue failed - 
48
    ✖ testExecuted, XCTUnwrap failed: expected non-nil value of type "TestSummary" -

**Comment by pepicrft** [comment]
> Any idea why two tests failed on Ubuntu? 😬

I recommend running them in Docker:

```bash
docker run --rm \
    --volume "$(pwd):/package" \
    --workdir "/package" \
    swift:5.5 \
    /bin/bash -c \
    "swift package fetch && swift test --build-path ./.build/linux"
```

If it fails consistently on CI it's possible it's a legit error due to implementations that are coupled to the OS.

**Comment by thii** [comment]
Thanks @joshdholtz!

**Pull request merged**